subprojects/packagefiles/c-ares/meson.build

```diff
@@ -0,0 +1,230 @@
+  cc.has_header('sys/uio.h') ? '-DHAVE_SYS_UIO_H' : [],
+
+  cc.has_type('struct timeval', prefix: '#include <sys/time.h>') ? '-DHAVE_STRUCT_TIMEVAL' : [],
+  cc.has_type('struct sockaddr_in6', prefix: '#include <sys/socket.h>\n#include <netinet/in.h>') ? '-DHAVE_STRUCT_SOCKADDR_IN6' : [],
+  cc.has_type('struct addrinfo', prefix: '#include <sys/types.h>\n#include <sys/socket.h>\n#include <netdb.h>') ? '-DHAVE_STRUCT_ADDRINFO' : [],
+
+  ## Wants meson 1.3.0
+  # cc.has_define('AF_INET6', prefix: '#include <sys/socket.h>\n#include <netinet/in.h>') ? '-DHAVE_AF_INET6' : [],
+  # cc.has_define('PF_INET6', prefix: '#include <sys/socket.h>\n#include <netinet/in.h>') ? '-DHAVE_PF_INET6' : [],
+  # cc.has_define('O_NONBLOCK', prefix: '#include <fcntl.h>') ? '-DHAVE_FCNTL_O_NONBLOCK' : [],
+
+  cc.has_function('recvfrom', prefix: '#include <sys/types.h>\n#include <sys/socket.h>') ? '-DHAVE_RECVFROM' : [],
+  cc.has_function('writev', prefix: '#include <sys/uio.h>') ? '-DHAVE_WRITEV' : [],
+
+  cc.links('#include <string.h>\nint main() {return (int)strncasecmp("a", "A", 1);}', name: 'strncasecmp') ? '-DHAVE_STRNCASECMP' : [],
+  cc.links('#include <string.h>\nint main() {return (int)strcasecmp("a", "A");}', name: 'strcasecmp') ? '-DHAVE_STRCASECMP' : [],
+  cc.links('#include <string.h>\nint main() {return (int)strcmpi("a", "A");}', name: 'strcmpi') ? '-DHAVE_STRCMPI' : [],
+  cc.links('#include <string.h>\nint main() {return (int)strncmpi("a", "A", 1);}', name: 'strncmpi') ? '-DHAVE_STRNCMPI' : [],
+  cc.links('#include <string.h>\nint main() {return (int)strnicmp("a", "A", 1);}', name: 'strnicmp') ? '-DHAVE_STRNICMP' : [],
+],
+'linux': [
+  '-DHAVE_RECV',
+    '-DRECV_TYPE_ARG1=int',
+    '-DRECV_TYPE_ARG2=void*',
+    '-DRECV_TYPE_ARG3=size_t',
+    '-DRECV_TYPE_ARG4=int',
+    '-DRECV_TYPE_RETV=ssize_t',
+  '-DHAVE_SEND',
+    '-DSEND_TYPE_ARG1=int',
+    '-DSEND_QUAL_ARG2=const',
+    '-DSEND_TYPE_ARG2=void*',
+    '-DSEND_TYPE_ARG3=size_t',
+    '-DSEND_TYPE_ARG4=int',
+    '-DSEND_TYPE_RETV=ssize_t',
+  '-DHAVE_AF_INET6',
+  '-DHAVE_PF_INET6',
+  '-DHAVE_FCNTL_O_NONBLOCK',
+],
+}
+cares_args_map += {'darwin': cares_args_map['linux']}
+
+cares_args = [cares_args_map['common'], cares_args_map.get(target_machine.system(), [])]
+
+cares_config = {
+'common': {
+  'CARES_HAVE_SYS_TYPES_H': cc.has_header('sys/types.h'),
+  'CARES_HAVE_SYS_RANDOM_H': cc.has_header('sys/random.h'),
+  'CARES_HAVE_SYS_SOCKET_H': cc.has_header('sys/socket.h'),
+  'CARES_HAVE_WINDOWS_H': cc.has_header('windows.h'),
+  'CARES_HAVE_WS2TCPIP_H': cc.has_header('ws2tcpip.h'),
+  'CARES_HAVE_WINSOCK2_H': cc.has_header('winsock2.h'),
+  'CARES_HAVE_ARPA_NAMESER_H': cc.has_header('arpa/nameser.h'),
+  'CARES_HAVE_ARPA_NAMESER_COMPAT_H': cc.has_header('arpa/nameser_compat.h'),
+  'CARES_TYPEOF_ARES_SSIZE_T': 'ssize_t',
+  'CARES_TYPEOF_ARES_SOCKLEN_T': 'socklen_t',
+},
+'windows': {
+  'CARES_TYPEOF_ARES_SSIZE_T': '__int64',
```

Comment:
intptr_t is typically used for ssize_t.

Reply:
Copied from upstream, but will check if intptr_t works